lib/msf/core/payload/windows/x64/addr_loader.rb

```diff
@@ -0,0 +1,83 @@
+# -*- coding: binary -*-
+
+require 'msf/core'
+require 'msf/core/payload/windows/x64/block_api'
+
+module Msf
+
+###
+#
+# Windows ARCH_X64 loader
+#
+###
+
+module Payload::Windows::AddrLoader_x64
+
+  include Msf::Payload::Windows
+  include Msf::Payload::Windows::BlockApi_x64
+
+  #
+  # Generate and compile the loader
+  #
+  def generate_loader
+    combined_asm = %Q^
+        cld                    ; Clear the direction flag.
+        and rsp, ~0xF          ;  Ensure RSP is 16 byte aligned
+        call start             ; Call start, this pushes the address of 'api_call' onto the stack.
+        #{asm_block_api}
+      start:
+        pop rbp
+        #{asm_block_loader}
+    ^
+    loader = Metasm::Shellcode.assemble(Metasm::X64.new, combined_asm).encode_string
+    offset_size = loader.index("AAAAAAAA")
+    offset_addr = loader.index("BBBBBBBB")
```

Comment:
Randomize?

Reply:
So these are just placeholders so we know which part of the shellcode to replace with the actual address and size (of the second stage / lpe exploit) at runtime. It doesn't make any sense to randomise them.